elm/index.js

```diff
@@ -1,7 +1,29 @@
+"use strict";
 import { Elm } from "./Main.elm";
 
-window.addEventListener('DOMContentLoaded', function () {
-  Elm.Main.init({
-    node: document.querySelector(".elm"),
+// Run Elm on all elm Nodes
+function initializeWidgets() {
+  const elmNodes = document.querySelectorAll(".elm");
+  elmNodes.forEach((node) => {
+    const app = Elm.Main.init({
+      node,
+      flags: getFlags(node.dataset.flags),
+    });
+    // Initialize ports below this line
   });
-})
+}
+
+// Parse the JSON from IHP or just pass null if there is no flags data
+function getFlags(data) {
+  return data ? JSON.parse(data) : null;
+}
+
+// Initialize Elm on page load
+window.addEventListener('load', (event) => {
```

Comment:
the rest of the code uses `"` instead of `'`

Reply:
Should be corrected now